ngraph/core/reference/src/runtime/reference/multiclass_nms.cpp

```diff
@@ -0,0 +1,514 @@
+
+                size_t max_num_of_selected_indices = selected_indices_shape[0];
+                size_t output_size = std::min(filteredBoxes.size(), max_num_of_selected_indices);
+
+                size_t idx;
+                for (idx = 0; idx < output_size; idx++)
+                {
+                    const auto& box_info = filteredBoxes[idx];
+                    SelectedIndex selected_index{box_info.batch_index, box_info.index, num_boxes};
+                    SelectedOutput selected_score{static_cast<float>(box_info.class_index),
+                                                  box_info.score,
+                                                  box_info.box.x1,
+                                                  box_info.box.y1,
+                                                  box_info.box.x2,
+                                                  box_info.box.y2};
+
+                    selected_indices_ptr[idx] = selected_index;
+                    selected_scores_ptr[idx] = selected_score;
+                }
+
+                SelectedIndex selected_index_filler{0, 0, 0};
+                SelectedOutput selected_score_filler{0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f};
+                for (; idx < max_num_of_selected_indices; idx++)
+                {
+                    selected_indices_ptr[idx] = selected_index_filler;
+                    selected_scores_ptr[idx] = selected_score_filler;
+                }
+            }
+
+            void multiclass_nms_postprocessing(void* prois,
+                                               void* pscores,
+                                               void* pselected_num,
+                                               const ngraph::element::Type output_type,
+                                               const std::vector<float>& selected_outputs,
+                                               const std::vector<int64_t>& selected_indices,
+                                               const std::vector<int64_t>& valid_outputs,
+                                               const ngraph::element::Type selected_scores_type)
+            {
+                auto num_selected = std::accumulate(valid_outputs.begin(), valid_outputs.end(), 0);
+
+                /* data */
+                size_t selected_outputs_size = num_selected * 6;
+
+                switch (selected_scores_type)
+                {
+                case element::Type_t::bf16:
+                {
+                    bfloat16* scores_ptr = static_cast<bfloat16*>(prois);
+                    for (size_t i = 0; i < selected_outputs_size; ++i)
+                    {
+                        scores_ptr[i] = bfloat16(selected_outputs[i]);
+                    }
+                }
+                break;
+                case element::Type_t::f16:
+                {
+                    float16* scores_ptr = static_cast<float16*>(prois);
+                    for (size_t i = 0; i < selected_outputs_size; ++i)
```

Comment:
Maybe `for (int64_t i = 0; i < selected_outputs_size; ++i)`?

Reply:
Done